File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/infrastructure/repository/QuestionOrderRepositoryImpl.java
package com.eager.questioncloud.core.domain.payment.infrastructure.repository;

import com.eager.questioncloud.core.domain.payment.infrastructure.entity.QuestionOrderEntity;
import com.eager.questioncloud.core.domain.payment.model.QuestionOrder;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class QuestionOrderRepositoryImpl implements QuestionOrderRepository {
    private final QuestionOrderJpaRepository questionOrderJpaRepository;

    @Override
    public void save(QuestionOrder questionOrder) {
        questionOrderJpaRepository.saveAll(QuestionOrderEntity.from(questionOrder));
    }

    @Override
    public void deleteAllInBatch() {
        questionOrderJpaRepository.deleteAllInBatch();
    }
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/payment/infrastructure/repository/QuestionOrderRepositoryImpl.kt
package com.eager.questioncloud.core.domain.payment.infrastructure.repository

import com.eager.questioncloud.core.domain.payment.infrastructure.entity.QuestionOrderEntity.Companion.from
import com.eager.questioncloud.core.domain.payment.model.QuestionOrder
import org.springframework.stereotype.Repository

@Repository
class QuestionOrderRepositoryImpl(
    private val questionOrderJpaRepository: QuestionOrderJpaRepository
) : QuestionOrderRepository {

    override fun save(questionOrder: QuestionOrder) {
        questionOrderJpaRepository.saveAll(from(questionOrder))
    }

    override fun deleteAllInBatch() {
        questionOrderJpaRepository.deleteAllInBatch()
    }
}
